flows/engine/session.go

```diff
@@ -101,14 +101,24 @@ func (s *session) Status() flows.SessionStatus { return s.status }
 func (s *session) Wait() flows.ActivatedWait   { return s.wait }
 
 func (s *session) CurrentContext() *types.XObject {
-	run := s.waitingRun()
+	run := s.currentRun()
 	if run == nil {
 		return nil
 	}
-
 	return types.NewXObject(run.RootContext(s.env))
 }
 
+// looks through this session's run for the one that was last modified
+func (s *session) currentRun() flows.FlowRun {
+	var lastRun flows.FlowRun
+	for _, run := range s.runs {
+		if lastRun == nil || run.ModifiedOn().After(lastRun.ModifiedOn()) {
```

Comment:
Just thinking aloud here. Last modified will be parent in the case of a child exiting and returning to parent?

Reply:
yeah we always go back thru the stack - added a subflow test to verify this